packages/rOpenScPCA/R/calculate-clusters.R

```diff
@@ -81,12 +86,19 @@ calculate_clusters <- function(
   # Check input arguments
   stopifnot(
     "`resolution` must be numeric" = is.numeric(resolution),
-    "`nn` must be numeric" = is.numeric(nn)
+    "`nn` must be numeric" = is.numeric(nn),
+    "`threads` must be numeric" = is.numeric(threads)
   )
 
   algorithm <- match.arg(algorithm)
   weighting <- match.arg(weighting)
-  objective_function <- match.arg(objective_function)
+
+  # this might be NA if it came from the sweep_clusters() function
+  if (is.na(objective_function)) {
+    objective_function <- NULL
+  } else {
+    objective_function <- match.arg(objective_function)
+  }
```

Comment:
On reflection, since this parameter only shows up in the output table if it is not `NA`, the previous solution should solve this just fine, and you can revert this change! I thought it would end up included with the default value, but we already don't do that because of the way we (properly!) handle cluster_args. (If you do want to keep it though, you will need to wrap `is.na` with an `all()` or check the length, I realized.)